cpp/src/graph.cpp

```diff
@@ -0,0 +1,67 @@
+#include "../include/graph.h"
+
+#include <fstream>
+#include <sstream>
+#include <iostream>
+
+using namespace std;
+
+ostream& operator<< (ostream& os, const Graph& graph) {
+    auto adj = graph.getAdj();
+
+    for (auto adjIter = adj.begin(); adjIter != adj.end(); ++adjIter) {
+        os<<adjIter->first<<" ";
+        for(auto neigIter = adjIter->second.begin(); neigIter != adjIter->second.end(); ++neigIter) {
+            os<<neigIter->v2<<" ";
+        }
+        os<<endl;
+    }
+
+    return os;
+}
+
+Graph::Graph(std::string fileName) {
+    deserialize(fileName);
+}
+
+void Graph::deserialize(std::string fileName) {
+    ifstream file(fileName);
+    string line;
+    while (getline(file, line)) {
+        istringstream iss(line);
+        int vertex, neighbour;
+
+        iss >> vertex;
+        this->adj[vertex] = vector<Edge>();
+        while (iss >> neighbour) {
+            this->adj[vertex].push_back(Edge(vertex, neighbour));
```

Comment:
using std::vector::emplace_back would be a little bit faster here (not that it matters much).

Calling a constructor directly as parameter creates an rvalue so we can use move mechanism instead of copying. See http://stackoverflow.com/a/4303834

```C++
std::vector<std::string> vec;
 vec.emplace_back(std::string("Hello")); // moves
 std::string s;
 vec.emplace_back(s); //copies
```